### PR TITLE
externalize config

### PR DESCRIPTION
**Pull request by ayushthe1**
Signed-off-by: ayush_gitk <ayushsharmaa101@gmail.com>

**Description**
externalize config and camelcase the keys in `metadata.json`

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: build/config.go [hunk 1]
```diff
@@ -60,41 +50,24 @@ func NewConfig(version string) manifests.Config {
...
+	if len(AllVersions) == 0 {
 		return
 	}
-	DefaultGenerationURL = "https://github.com/nginxinc/helm-charts/blob/master/stable/nginx-service-mesh-" + version + ".tgz?raw=true"
-	DefaultGenerationMethod = adapter.HelmCHARTS
-	WorkloadPath = oam.WorkloadPath
-	MeshModelPath = filepath.Join(wd, "templates", "meshmodel", "components")
+	LatestVersion = AllVersions[len(AllVersions)-1]
+	DefaultGenerationMethod = adapter.Manifests
+	DefaultGenerationURL = "https://raw.githubusercontent.com/nginxinc/nginx-service-mesh/" + LatestVersion + ""
```
Comment (ayushthe1) [inline build/config.go:72]: I have confusion on which file to reference for `DefaultGenerationURL`. Could anyone point it out.

File: build/meshmodel_metadata.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+	"primaryColor":   "#009639",
+	"secondaryColor": "#42C473",
+	"shape":           "circle",
+	"logoUrl":        "https://www.vectorlogo.zone/logos/nginx/index.html",
+	"svgColor":       "",
```
Comment (Revolyssup) [inline build/meshmodel_metadata.json:6]: Why is the svgColor empty?

File: build/config.go (hunk 1 above)
Comment (Revolyssup) [reply build/config.go:72]: Use `DefaultGenerationMethod` as  `adapter.HelmCHARTS` along with previous `DefaultGenerationURL`.
nginxinc/nginx can be used with utils.GetLatestTagSorted to get the latest version and when we have the latest version then the older way of generation and URL can be used.

**Review by Revolyssup — CHANGES_REQUESTED** [review]

**Comment by ayushthe1** [comment]
@Revolyssup made the changes as you pointed out but i can't figure out why the e2e tests are failing.

**Review by Revolyssup — APPROVED** [review]

**Pull request merged**